src/participants/get_src_state.py

```diff
@@ -0,0 +1,108 @@
+# Copyright (C) 2017 Jolla Ltd.
+# All rights reserved.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.
+
+"""Check to see the source state for a particular package
+   Success if there is src ready to build
+   service_state has more details
+
+   Packages which don't use a service are indistinguishable from a
+   successful service run.
+
+:term:`Workitem` fields IN:
+
+:Parameters:
+   :ev.namespace (string):
+      Used to contact the right OBS instance.
+   :package (string):
+      Package name to be checked
+   :project (string):
+      OBS project in which the package lives
+
+
+:term:`Workitem` params IN
+
+:Parameters:
+   :package (string):
+      Package name to be checked, overrides the package field
+   :project (string):
+      OBS project in which the package lives, overrides the project field
```

Comment:
in a process there is a workitem which has a fields key. That is passed through all participants and they can modify it as it flows. These are field values
In the process definition the particpant can be followed by a named parameter/value : 
  get_source :project => "home:lbt"
These are parameter values.
Because these are hard set in the process definition they take precedence. The process doesn't want that information flowing in via the fields (the pdef could choose to initialise field values and allow them to be overriden)